Tests/src/PublicAPITests/HotspotSequenceTests.cpp

```diff
@@ -689,4 +692,400 @@ CSP_PUBLIC_TEST(CSPEngine, HotspotSequenceTests, GenerateSequenceKeyTest)
+	LogIn(UserSystem, UserId);
+
+	// Create space
+	csp::systems::Space Space;
+	CreateSpace(SpaceSystem, UniqueSpaceName, TestSpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, Space);
+
+	auto [EnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	EXPECT_EQ(EnterResult.GetResultCode(), csp::systems::EResultCode::Success);
+
+	EntitySystem->SetEntityCreatedCallback(
+		[](csp::multiplayer::SpaceEntity* Entity)
+		{
+		});
+
+	// Create object to represent the hotspot
+	csp::common::String ObjectName					 = "Object 1";
+	csp::multiplayer::SpaceTransform ObjectTransform = {csp::common::Vector3::Zero(), csp::common::Vector4::Zero(), csp::common::Vector3::One()};
+	auto [CreatedObject]							 = AWAIT(EntitySystem, CreateObject, ObjectName, ObjectTransform);
+
+	bool ComponentAdded = false;
+
+	CreatedObject->SetUpdateCallback(
+		[&ComponentAdded, ObjectName](csp::multiplayer::SpaceEntity* Entity,
+									  csp::multiplayer::SpaceEntityUpdateFlags Flags,
+									  csp::common::Array<csp::multiplayer::ComponentUpdateInfo>& UpdateInfo)
+		{
+			if (Entity->GetName() == ObjectName)
+			{
+				for (size_t i = 0; i < UpdateInfo.Size(); ++i)
+				{
+					if (UpdateInfo[i].UpdateType == csp::multiplayer::ComponentUpdateType::Add)
+					{
+						ComponentAdded = true;
+					}
+				}
+			}
+		});
+
+	// Create hotspot component
+	auto* HotspotComponent
+		= static_cast<csp::multiplayer::HotspotSpaceComponent*>(CreatedObject->AddComponent(csp::multiplayer::ComponentType::Hotspot));
+
+	CreatedObject->QueueUpdate();
+	WaitForCallbackWithUpdate(ComponentAdded, EntitySystem);
+
+	EXPECT_TRUE(ComponentAdded);
+
+	// Create Hotspot groups
+	csp::common::String TestGroupName1 = "CSP-UNITTEST-SEQUENCE-MAG1";
+	csp::common::String TestGroupName2 = "CSP-UNITTEST-SEQUENCE-MAG2";
+	csp::common::String TestItemName   = "AnotherItem";
+
+	{
+		// Create 2 groups that contains the component
+
+		// Create one with only a single item to test deletion functionality
+		csp::systems::HotspotGroup HotspotGroup1;
+		CreateHotspotgroup(HotspotSystem, TestGroupName1, {HotspotComponent->GetUniqueComponentId()}, HotspotGroup1);
+
+		// Create one with an additional item to test update functionality
+		csp::systems::HotspotGroup HotspotGroup2;
+		CreateHotspotgroup(HotspotSystem, TestGroupName2, {HotspotComponent->GetUniqueComponentId(), TestItemName}, HotspotGroup2);
+
+		// Ensure the 2 groups are created correctly
+		csp::common::Array<csp::systems::HotspotGroup> FoundGroups;
+		GetHotspotGroups(HotspotSystem, {TestGroupName1, TestGroupName2}, FoundGroups);
+
+		EXPECT_EQ(FoundGroups.Size(), 2);
+	}
+
+	// Remove entity
+	{
+		bool EntityRemoved = false;
+
+		CreatedObject->Destroy(
+			[&EntityRemoved](bool Success)
+			{
+				EntityRemoved = true;
+			});
+
+		WaitForCallbackWithUpdate(EntityRemoved, EntitySystem);
+
+		EXPECT_TRUE(EntityRemoved);
+
+		std::this_thread::sleep_for(2500ms);
```

Comment:
(See the other comment suggesting awaiting the sequence delete multiplayer event.)